ControlRoom/Controllers/LocalSearchController.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  LocalSearchController.swift
+//  ControlRoom
+//
+//  Created by John McEvoy on 29/11/2023.
+//  Copyright © 2023 Paul Hudson. All rights reserved.
+//
+
+import Foundation
+import MapKit
+
+@MainActor
+class LocalSearchController: NSObject, ObservableObject {
+    /// Prevents duplicate queries from being made
+    private var lastQuery: String = ""
+
+    /// Completion handler is called by the `MKLocalSearchCompleter` success callback
+    private var callback: (([LocalSearchResult]) -> Void)?
+
+    /// the MKLocalSearchCompleter used to make local search requests
+    private lazy var localSearchCompleter: MKLocalSearchCompleter = {
+        let completer = MKLocalSearchCompleter()
+        completer.resultTypes = [.address, .pointOfInterest]
+        completer.delegate = self
+        return completer
+    }()
+
+    /**
+     Finds places and POIs using a query string and a geographical point to focus on.
+
+     - Parameter for: The partial (autocomplete) query to search for.
+     - Parameter around: Provides a hint for `MKLocalSearchCompleter` to search around a geographical point.
+     - Parameter completion: Called if valid search results are found.
+
+     - Returns: If a location is found immediately (a coordinate was pasted in, for example), returns a `Location`.
+     */
+    func search(for query: String,
+                around location: Location,
+                completion: @escaping ([LocalSearchResult]) -> Void) -> Location? {
```

Comment:
```suggestion
    func search(
                        for query: String,
                       around location: Location,
                       completion: @escaping ([LocalSearchResult]) -> Void
                       ) -> Location? {
```

Now you have ctrl+M shortcut for make this

Reply:
Sorry @Harry-KNIGHT I don't see any changes to this code, just some re-formatting. What are you looking to change?